lib/pages/tabs/settings_tab.dart

```diff
@@ -0,0 +1,257 @@
+                  maxWidth: 640,
+                ),
+                child: SailColumn(
+                  spacing: SailStyleValues.padding50,
+                  children: [
+                    NodeConnectionSettings(
+                      name: 'Sidechain',
+                      connected: viewModel.sidechainConnected,
+                      settings: viewModel.sidechainSettings,
+                      testConnectionValues: viewModel.reconnectSidechain,
+                      connectionError: viewModel.sidechainConnectionError,
+                      loading: viewModel.sidechainBusy,
+                    ),
+                    NodeConnectionSettings(
+                      name: 'Mainchain',
+                      connected: viewModel.mainchainConnected,
+                      settings: viewModel.mainchainSettings,
+                      testConnectionValues: viewModel.reconnectMainchain,
+                      connectionError: viewModel.mainchainConnectionError,
+                      loading: viewModel.mainchainBusy,
+                    ),
+                  ],
+                ),
+              ),
+              Expanded(child: Container()),
+            ],
+          );
+        }),
+      ),
+    );
+  }
+}
+
+class NodeConnectionSettings extends ViewModelWidget<NodeConnectionViewModel> {
+  final String name;
+  final bool connected;
+  final SingleNodeConnectionSettings settings;
+  final VoidCallback testConnectionValues;
+  final String? connectionError;
+  final bool loading;
+
+  const NodeConnectionSettings({
+    super.key,
+    required this.name,
+    required this.connected,
+    required this.settings,
+    required this.testConnectionValues,
+    required this.connectionError,
+    required this.loading,
+  });
+
+  @override
+  Widget build(BuildContext context, NodeConnectionViewModel viewModel) {
+    final theme = SailTheme.of(context);
+    return SailColumn(
+      spacing: SailStyleValues.padding25,
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        SailColumn(
+          spacing: SailStyleValues.padding08,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            SailRow(
+              spacing: SailStyleValues.padding08,
+              children: [
+                SailText.primary15(name),
+                SailErrorShadow(
+                  enabled: !connected,
+                  child: SailSVG.fromAsset(
+                    SailSVGAsset.iconGlobe,
+                    width: SailStyleValues.iconSizePrimary,
+                    color: connected ? theme.colors.success : theme.colors.error,
+                  ),
+                ),
+              ],
+            ),
+            if (connectionError != null) SailText.secondary12(connectionError!),
+          ],
+        ),
+        SailTextField(
+          label: 'Config path',
+          controller: settings.configPathController,
+          hintText: '/the/path/to/your/somethingchain.conf',
```

Comment:
Could this be a file explorer input? Dunno how that works in Flutter

Reply:
ill add it to the issue tracker